### PR TITLE
Refactor with Yarn Workspaces

### PR DESCRIPTION
**Pull request by joebochill**
#### Changes proposed in this Pull Request:
- Reorganize the repository
    - moved packages into a subfolder
    - moved demo to root
- Integrate yarn workspaces and remove linking scripts
    - Simplifies the build commands (every project has its own build command and running build in the root executes them all in the correct order based on dependencies)
- Convert mui package to build using tsc and keep a copy of the tsx components for easier reviewing
    - this will let us see if there are any anomalies in the build output (before we transpiled everything and never checked in the built built files so you could never see what the result of the build is)
- Use all icon packages in the demo project (font, svg, mui)

> TODO: The docs and CI stuff will need to be cleaned up to reflect the new repository structure / commands. External links coming in from places like doc-it should also probably be checked.

#### To Test:
- Read the readme / set up some time with me

### CODE REVIEW

**Comment by joebochill** [comment]
I went ahead and tackled the merge conflicts — they weren't too bad, but it definitely helped being very familiar with what both of you had already done. We just need to do a read through of all of the documentation to make sure everything is still correct (And identify if there are any places/links on Doc-it that we will need to update).

**Comment by daileytj** [comment]
@joebochill thanks for looking at this. Should I wait for this before publishing icons-mui@v3.3.0 ?

**Comment by daileytj** [comment]
@joebochill @bkarambe are we still waiting on the doc/README updates? And should the finite set of svg's be setup now so the demo doesn't just show 1000 of the same icon? (If that is out of scope feel free to say so)

**Comment by joebochill** [comment]
> @joebochill @bkarambe are we still waiting on the doc/README updates? And should the finite set of svg's be setup now so the demo doesn't just show 1000 of the same icon? (If that is out of scope feel free to say so)

Yeah, we might as well (should only take 3 minutes) update that to just show a selection of icons (maybe 10-15) including:
- Eaton logos (3)
- EV plug icons with the overridden name format (2)
- A couple normal icons (from the beginning, middle, and end of the alphabet)

**Comment by bkarambe** [comment]
> I haven't seen a push or comment about the docs, so I assume that is still not done. @bkarambe

No @joebochill This is still not done. I will update the PR soon

**Comment by huayunh** [comment]
As a reminder, all icon download links on docit will be broken as soon as this gets merged to master. Make sure you modify docit's icon page before this icons repo dev-to-master merge.

File: packages/icon-font/README.md
Comment (joebochill) [inline packages/icon-font/README.md:28]: These changes are incorrect — the packages folder will never appear in a node_modules folder.

The correct path now will be `./node_modules/@brightlayer-ui/icons/BrightlayerUIIcons.css`.

This will necessitate a new major version on the package because the import path has changed.

File: packages/mui/README.md
Comment (joebochill) [inline packages/mui/README.md:66]:
```suggestion
This will populate the /dist folder with all of the icon components as well as copy over necessary files for publishing (Readme, package.json, etc.).
```

**Comment by joebochill** [comment]
The files that need to be updated are the following:

- **/.circleci/config.yml** (publish job needs to be updated to publish icon-font from the correct folder, not from the root...see comment below about adding the command in icon font package.json file)
- **/README.md** (add a small blurb in the maintainers section that mentions the `build` command between the install and start blurbs)
- **/PUBLISHING.md** (this file needs to be split and moved into the /packages/icon-font and /packages/svg folders)
- **/VALIDATION.md** (need to be updated to reflect the new demo project instead of the showcase and include Icons SVG information.)
- **/packages/icon-font/package.json** (needs publish and tag commands in the scripts object, needs major version increment and changelog to reflect updated import path)
- **/packages/icon-font/README.md** (update paths per the existing comments on this PR)
- **/packages/mui/README.md** (update paths per the existing comments on this PR)
- **/packages/mui/PUBLISHING.md** (update build instructions to reflect only a single build command from the root is necessary)
- **/packages/rn-vector/README.md** (update paths per the existing comments on this PR)
- **/packages/svg/README.md** (there are duplicate readmes in this folder...consolidate them and make sure the resulting file is correct)
**/demos/react-demo/README.md** (might be nice to preface this with a paragraph that explains what this demo is and how it works before the CRA boilerplate stuff)
**/demos/react-demo/src/pages/svg.tsx** (update to show a handful of SVG icons only per the earlier PR comment)

Looks like all of the packages also need to have the license crawler scripts/functionality re-added (we can probably make a new issue for this to avoid the scope creep...please write one).

**Comment by bkarambe** [comment]
> license crawler scripts/functionality

Added a feature request for license crawler https://github.com/brightlayer-ui/icons/issues/251

**Review by bkarambe — APPROVED** [review]

**Review by jeffvg — APPROVED** [review]
everything installs, builds and runs. Packages look good

**Pull request merged**
